string path
int16 flag
---
